Xamarin.Forms.Platform.Android/AppCompat/Platform.cs

```diff
@@ -267,16 +267,17 @@ bool HandleBackPressed(object sender, EventArgs e)
 			return handled;
 		}
 
-		void LayoutRootPage(Page page, int width, int height)
+		internal static void LayoutRootPage(FormsAppCompatActivity activity, Page page, int width, int height)
 		{
-			var activity = (FormsAppCompatActivity)_context;
 			int statusBarHeight = Forms.IsLollipopOrNewer ? activity.GetStatusBarHeight() : 0;
+			statusBarHeight = activity.Window.Attributes.Flags.HasFlag(WindowManagerFlags.Fullscreen) ? 0 : statusBarHeight;
+			statusBarHeight = Forms.TitleBarVisibility == AndroidTitleBarVisibility.Never ? 0 : statusBarHeight; // just to be sure
```

Comment:
A single ternary statement might also be better...it just seems easier to follow the logic if it's one statement instead of two.